MoreLinq/Memoize.cs

```diff
@@ -30,6 +30,10 @@ static partial class MoreEnumerable
         /// </summary>
         /// <param name="source">The source sequence.</param>
         /// <returns>Returns a sequence that corresponds to a cached version of the input sequence.</returns>
+        /// <remarks>
+        /// Enumeration of the result sequence is thread-safe, that is, multiple independent IEnumerator values 
```

Comment:
Being thread-safe is different from being thread-agnostic. Being thread-safe means generally that code is safe in face of concurrency and parallelism; it can be called from multiple thread _at the same time_. Thread-agnostic means code has no association, link or affinity to a particular thread. It is safe to call it from different threads, just not at the same time. Of course, thread-safe is thread-agnostic but the reverse is not always true.

> also could you help me with remarks? i am not so good with documentation.

Yes, okay, I'll propose something soon.